vmss-prototype/vmss-prototype

```diff
@@ -905,17 +905,14 @@ def in_cluster(sub_args, func):
                     '--tenant', tenant,
                     '--service-principal'
                     ]
+        if username == "msi":
+            login_cmd = ['az', 'login', '--identity']
         masq = [username, password, tenant]
 
         # We have seen temporary network failures cause this to fail so we will
         # add a simple retry loop here, with slowly increasing delay between
         # each retry.
-        run(['az', 'login',
-             '--username', username,
-             '--password', password,
-             '--tenant', tenant,
-             '--service-principal'
-             ], retries=10, masq=masq, retry_wait_duration_func=lambda num_retry: 5 * num_retry + 1)
+        run(login_cmd, retries=10, masq=masq, retry_wait_duration_func=lambda num_retry: 5 * num_retry + 1])
```

Comment:
Is this `run` `masq` assignment going to work O.K. in the MSI scenario? (in other words, when the `run` command doesn't actually include the `username`, `password`, or `tenant` args?

Reply:
Yes, the masq is a list of things to masq from the log - sensitive data.
It does not are if it is there or not - it just makes sure that the log does not include it.

Reply:
PS - this is my personal account (separate from my Microsoft identity)